### ISSUE TITLE
BigDecimal#percent_of

### ISSUE THREAD

**Issue by mikeymicrophone** [readysteady/percentage#2]
I'd like to be able to use decimals this way.  However, I note that the library treats a decimal value between 0 and 1 as a percentage of 1.  I'd prefer to keep it as a scale of 0 to 100.  When I say "0.5 percent of 88", I don't mean 50 percent.

What do you think

**Comment by timcraft** [comment]
The simplest way to get the correct semantics would be to copy Integer#percent_of but use `Percentage(self)` instead of `Percentage.new` to convert the decimal to a fractional value.

I think the reason I didn't add that method originally was because it doesn't read as well with decimals: `10.percent_of(100)` reads like english but `BigDecimal('0.5').percent_of(88)` doesn't.

It's just syntactic sugar for multiplication, so it would help to understand what kind of naming you're using. Do you have a simplified snippet of the code you're writing that would be clearer if this method was added? With variables like `decimal.percent_of(number)` I'd be inclined to rewrite that to use multiplication directly like `number * Percentage(decimal)`

**Comment by mikeymicrophone** [comment]
The context where this jumped out to me is working with db columns that are decimals in Rails.
```
deal.agent_split_percentage.percent_of deal.inbound_commission
```

Thanks for the instructions.  I did spend time in implementation but I didn't get all the way to the bottom of the distinction between the two constructor syntaxes.

**Comment by timcraft** [comment]
@mikeymicrophone Thanks. Adding some #to_percentage methods might help as well, then you'd have the option of either #percent_of or something like this:

```ruby
deal.agent_split.to_percentage * deal.inbound_commission
```

No objections to adding BigDecimal#percent_of anyway... if you can put together a pull request with specs I can get that merged and released 🔜

**Comment by timcraft** [comment]
I've merged the PR and made a few tweaks, available in v1.1.0

Both of those examples should now work 😁
